Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -332,54 +332,32 @@ public IAsyncLoop Mine(WalletSecret walletSecret)
             this.logger.LogTrace("()");
 
             BlockTemplate blockTemplate = null;
-            bool tryToSync = true;
 
             while (!this.nodeLifetime.ApplicationStopping.IsCancellationRequested)
             {
-                ChainedBlock chainTip = this.chain.Tip;
-                if (chainTip != this.consensusLoop.Tip)
-                {
-                    this.logger.LogTrace("(-)[SYNC_OR_REORG]");
-                    return;
-                }
-
                 while (!this.connection.ConnectedNodes.Any() || this.chainState.IsInitialBlockDownload)
                 {
                     if (!this.connection.ConnectedNodes.Any()) this.logger.LogTrace("Waiting to be connected with at least one network peer...");
                     else this.logger.LogTrace("Waiting for IBD to complete...");
 
-                    tryToSync = true;
                     await Task.Delay(TimeSpan.FromMilliseconds(this.minerSleep), this.nodeLifetime.ApplicationStopping).ConfigureAwait(false);
                 }
 
-                // TODO: What is the purpose of this conditional block?
-                // It seems that if it ends with continue, the next round of the loop will just 
-                // not execute the above while loop and it won't even enter this block again,
-                // so it seems like no operation.
-                // In StratisX there is the wait uncommented. Then it makes sense. Why we have it commented?
-                if (tryToSync)
```

Comment:
Why remove this code?
This will wait for at least 3 peers  before staking starts (I guess it so we have a better time sync sample)

Reply:
OK added a comment, I don't see much value in waiting for 3 nodes (apart for the time sync) but then again there is no downside (apart for waiting a bit longer to stake) in connecting to some more nodes.

So I am unsure about this.

Reply:
Any code we have in the code base should be doing something useful. We can wait longer, but what is the benefit of it? If we found a block instead and sent it to the network, we could earn some money. Instead we are waiting. The reason we are waiting is probably to get a better chance for our block to be accepted. But at the same time, if we are not looking for it, our chance of getting reward 0.

Time syncing is another topic I think and just by connected to 3 nodes it won't trigger. The threshold is little higher than that (and we also intend to have an option to have it disabled). However, the worst case scenario is that we have wrong time and we mine a new block with wrong time stamp. In worst case it would be rejected. 

To summarize, I can't see enough justification for this code, so I removed it.

Reply:
I'm happy with this.

Reply:
hmf... ok